### PR TITLE
Implement undo add and remove from reading list feature

### PR DESCRIPTION
**Pull request by aceTmobile**
- user can undo action when clicking the "Want to Read" button in the books grid view.
- user can undo action when clicking the "Remove" button in the reading list view.
- implemented e2e test for undo feature

### CODE REVIEW

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (catfireparty) [inline apps/okreads-e2e/src/integration/reading-list.spec.ts]: Why not just use `.first()`?

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (catfireparty) [inline apps/okreads-e2e/src/integration/reading-list.spec.ts]: There is a way to do this without jumping outside of Cypress into JQuery. Look into using closures -> https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Closures

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (catfireparty) [inline apps/okreads-e2e/src/integration/reading-list.spec.ts]: You can't really do this in Cypress. It may work, but it's probably not doing what you expect.

cf. https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Return-Values

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (catfireparty) [inline apps/okreads-e2e/src/integration/reading-list.spec.ts]: Don't do this in Cypress. Work with the Cypress way of writing tests rather than trying to make things synchronous.

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (catfireparty) [inline apps/okreads-e2e/src/integration/reading-list.spec.ts]: Look into Cypress assertions instead -> https://example.cypress.io/commands/assertions#should

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (catfireparty) [inline apps/okreads-e2e/src/integration/reading-list.spec.ts]: DRY?

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (catfireparty) [inline apps/okreads-e2e/src/integration/reading-list.spec.ts]: cf. other comments about working _with_ Cypress not against it.

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (catfireparty) [inline apps/okreads-e2e/src/integration/reading-list.spec.ts]: `.last()`?

File: libs/books/feature/src/lib/book-search/book-search.component.ts
Comment (catfireparty) [inline libs/books/feature/src/lib/book-search/book-search.component.ts]: We don't typically apply underscores to private variables. See previously defined variables.

File: libs/books/feature/src/lib/book-search/book-search.component.ts
Comment (catfireparty) [inline libs/books/feature/src/lib/book-search/book-search.component.ts]: Is this the right place to do this? What if the API call fails?

File: libs/books/feature/src/lib/book-search/book-search.component.ts
Comment (catfireparty) [inline libs/books/feature/src/lib/book-search/book-search.component.ts]: Is that long enough for a partially sighted user to find the snackbar?

File: libs/books/feature/src/lib/reading-list/reading-list.component.ts
Comment (catfireparty) [inline libs/books/feature/src/lib/reading-list/reading-list.component.ts]: You've got roughly the same code in two places now. Could you consolidate this into an effect, perhaps?

**Review by catfireparty — CHANGES_REQUESTED** [review]

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (aceTmobile) [reply apps/okreads-e2e/src/integration/reading-list.spec.ts]: done

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (aceTmobile) [reply apps/okreads-e2e/src/integration/reading-list.spec.ts]: done

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (aceTmobile) [reply apps/okreads-e2e/src/integration/reading-list.spec.ts]: updated

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (aceTmobile) [reply apps/okreads-e2e/src/integration/reading-list.spec.ts]: updated

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (aceTmobile) [reply apps/okreads-e2e/src/integration/reading-list.spec.ts]: done

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (aceTmobile) [reply apps/okreads-e2e/src/integration/reading-list.spec.ts]: done

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (aceTmobile) [reply apps/okreads-e2e/src/integration/reading-list.spec.ts]: updated

File: apps/okreads-e2e/src/integration/reading-list.spec.ts
Comment (aceTmobile) [reply apps/okreads-e2e/src/integration/reading-list.spec.ts]: done

File: libs/books/feature/src/lib/book-search/book-search.component.ts
Comment (aceTmobile) [reply libs/books/feature/src/lib/book-search/book-search.component.ts]: updated

File: libs/books/feature/src/lib/book-search/book-search.component.ts
Comment (aceTmobile) [reply libs/books/feature/src/lib/book-search/book-search.component.ts]: updated

File: libs/books/feature/src/lib/book-search/book-search.component.ts
Comment (aceTmobile) [reply libs/books/feature/src/lib/book-search/book-search.component.ts]: updated

File: libs/books/feature/src/lib/reading-list/reading-list.component.ts
Comment (aceTmobile) [reply libs/books/feature/src/lib/reading-list/reading-list.component.ts]: updated

**Comment by mandarini** [comment]
Please resolve conflicts, rebase with chore/code-review, squash commits!

**Review by mandarini — CHANGES_REQUESTED** [review]
Please resolve conflicts, rebase with chore/core-review, squash commits!

**Comment by aceTmobile** [comment]
@mandarini Hi, I rebased with chore/code-review. Please re-review, all conflicts resolved. Question on squashing commits, I typically hit the squash and merge button above when merging into src branch to squash commits. Is this ok or are you wanting the commits squashed before you approve?

**Comment by aceTmobile** [comment]
@mandarini Hi I was able to squash commits but ran into conflicts after adding spacing. Does this work? If no, then yes I definitely need a little help on this thanks.

**Comment by aceTmobile** [comment]
@mandarini  hi, I made corrections. The mr should be good now. Please re-review. Thanks.

**Comment by mandarini** [comment]
Looks good to me

**Pull request merged**
